src/components/graph/GraphAttachedItemGroup.tsx

```diff
@@ -0,0 +1,192 @@
+/** ****************************************************************************
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************** */
+
+import moment from 'moment';
+import React from 'react';
+import { motion, AnimatePresence } from 'framer-motion';
+import { getTimestampAsNumber } from '../../helpers/date';
+import { createBemElement } from '../../helpers/styleCreators';
+import useOutsideClickListener from '../../hooks/useOutsideClickListener';
+import { EventAction } from '../../models/EventAction';
+import { EventMessage } from '../../models/EventMessage';
+import { ModalPortal } from '../Portal';
+import { isEqual } from '../../helpers/object';
+import { AttachedItem } from './GraphChunk';
+
+interface GraphAttachedItemProps {
+	items: AttachedItem[];
+	leftPosition: number;
+	bottomPosition: number;
+	isInfoOpen: boolean;
+	setExpandedItem: (item: EventAction | EventMessage | null) => void;
+	className?: string;
+}
+
+const GraphAttachedItem = ({
+	items,
+	leftPosition,
+	bottomPosition,
+	isInfoOpen,
+	setExpandedItem,
+	className,
+}: GraphAttachedItemProps) => {
+	const itemRef = React.useRef<HTMLDivElement>(null);
+	const itemGlobalPosition = React.useMemo(() => {
+		if (!itemRef.current) {
+			return {
+				left: 0,
+				top: 0,
+			};
+		}
+
+		const { left, top } = itemRef.current.getBoundingClientRect();
+
+		return {
+			left,
+			top,
+		};
```

Comment:
having itemRef.current?.getBoundingClientRect() in dependecies makes no sence, this wont trigger rerender and it always returns a new object that will never equal previous